### DISCUSSION TITLE
Creating a magnetic field without defining a coil

### DISCUSSION THREAD

**Discussion by eloisewm** [bfieldtools/bfieldtools#57] [Q&A] [answered]
This package seems amazing and is suuuper relevant to my work in MRI! However I'm very much a newbie at python and am currently trying to understand all of the applications of it. I'm looking to immerse a conductive cylinder into various magnetic fields and measure the resulting eddy currents - is there any way to impose a background field without defining a corresponding coil? 

For simple fields I can imagine I'd be able to construct some simple coils that would get the job done, but down the line I'm dealing with fields that are quite involved.

Any insight is very very much appreciated!
Eloise

**Comment by anttimc** [comment, marked as answer]
Hi Eloise,

sorry for answering this late (all the original developers work on other projects now). 

You can have any external field inducing the eddy currents. There's no need to specify a coil for that. The external field produces an EMF on the conducting surface. You get a linear system of equations for the time evolution of the eddy currents. It's a mess unless decouple those time-evolving currents patterns by the eigenvalue decomposition of the inductance matrix of the (discretized) conductor. An example of this [here](https://github.com/bfieldtools/bfieldtools/blob/master/examples/coil_design/minimal_eddy_current_cylindrical_coil_design.py) on line 153.

You need to map the code in the example to the equations that you can find for example the Zevenhoven paper in the [bfieldtools literature page](https://bfieldtools.github.io/literature.html). So you need to understand the theory and code a bit work out your problem. In the end, the solution itself should be just a few lines of python code.

-Antti
